lib/spack/spack/build_environment.py

```diff
@@ -236,50 +236,46 @@ def set_compiler_environment_variables(pkg, env):
 
 
 def set_build_environment_variables(pkg, env, dirty=False):
-    """
-    This ensures a clean install environment when we build packages.
+    """Ensure a clean install environment when we build packages.
 
-    Arguments:
-    dirty -- skip unsetting the user's environment settings.
-    """
-    # Add spack build environment path with compiler wrappers first in
-    # the path. We add both spack.env_path, which includes default
-    # wrappers (cc, c++, f77, f90), AND a subdirectory containing
-    # compiler-specific symlinks.  The latter ensures that builds that
-    # are sensitive to the *name* of the compiler see the right name
-    # when we're building with the wrappers.
-    #
-    # Conflicts on case-insensitive systems (like "CC" and "cc") are
-    # handled by putting one in the <build_env_path>/case-insensitive
-    # directory.  Add that to the path too.
-    env_paths = []
-    compiler_specific = join_path(spack.build_env_path, pkg.compiler.name)
-    for item in [spack.build_env_path, compiler_specific]:
-        env_paths.append(item)
-        ci = join_path(item, 'case-insensitive')
-        if os.path.isdir(ci):
-            env_paths.append(ci)
-
-    env_paths = filter_system_paths(env_paths)
+    This involves unsetting pesky environment variables that may
+    affect the build. It also involves setting environment variables
+    used by Spack's compiler wrappers.
 
-    for item in reversed(env_paths):
-        env.prepend_path('PATH', item)
-    env.set_path(SPACK_ENV_PATH, env_paths)
+    :param pkg: The package we are building
+    :param env: The build environment
+    :param bool dirty: Skip unsetting the user's environment settings
```

Comment:
OT: @alalazo @adamjstewart @davydden: would anyone object to using the [Google docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) instead of the [rest format](https://stackoverflow.com/questions/3898572/what-is-the-standard-python-docstring-format)?  I find it much more readable, and Sphinx understands it as well.